{
  "description": "sample Kubernetes/kb_1024_Normalized.json",
  "meta": {
    "full_size": 2916,
    "stripped_size": 643,
    "features": [
      "$ref",
      "format",
      "format:int32",
      "items"
    ],
    "raw_features": [
      "definitions",
      "properties",
      "type",
      "type:[]",
      "type:array",
      "type:integer",
      "type:null",
      "type:object",
      "type:string"
    ]
  },
  "schema": {
    "definitions": {
      "https:__kubernetesjsonschema_dev_master__defs_json_defs_io_k8s_apimachinery_pkg_apis_meta_v1_StatusCause": {
        "description": "StatusCause provides more information about an api.Status failure, including cases when multiple errors are encountered.",
        "properties": {
          "field": {
            "description": "The field of the resource that has caused this error, as named by its JSON serialization. May include dot and postfix notation for nested attributes. Arrays are zero-indexed.  Fields may appear more than once in an array of causes due to fields having multiple errors. Optional.\n\nExamples:\n  \"name\" - the field \"name\" on the current resource\n  \"items[0].name\" - the field \"name\" on the first array entry in \"items\"",
            "type": "string"
          },
          "message": {
            "description": "A human-readable description of the cause of the error.  This field may be presented as-is to a reader.",
            "type": "string"
          },
          "reason": {
            "description": "A machine-readable description of the cause of the error. If this value is empty there is no information available.",
            "type": "string"
          }
        },
        "type": "object"
      }
    },
    "description": "StatusDetails is a set of additional properties that MAY be set by the server to provide additional information about a response. The Reason field of a Status object defines what attributes will be set. Clients must ignore fields that do not match the defined type of each attribute, and should assume that any attribute may be empty, invalid, or under defined.",
    "properties": {
      "causes": {
        "description": "The Causes array includes more details associated with the StatusReason failure. Not all StatusReasons may provide detailed causes.",
        "items": {
          "$ref": "#/definitions/https:__kubernetesjsonschema_dev_master__defs_json_defs_io_k8s_apimachinery_pkg_apis_meta_v1_StatusCause"
        },
        "type": [
          "array",
          "null"
        ]
      },
      "group": {
        "description": "The group attribute of the resource associated with the status StatusReason.",
        "type": [
          "string",
          "null"
        ]
      },
      "kind": {
        "description": "The kind attribute of the resource associated with the status StatusReason. On some operations may differ from the requested resource Kind. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds",
        "type": [
          "string",
          "null"
        ]
      },
      "name": {
        "description": "The name attribute of the resource associated with the status StatusReason (when there is a single name which can be described).",
        "type": [
          "string",
          "null"
        ]
      },
      "retryAfterSeconds": {
        "description": "If specified, the time in seconds before the operation should be retried. Some errors may indicate the client must take an alternate action - for those errors this field may indicate how long to wait before taking the alternate action.",
        "format": "int32",
        "type": [
          "integer",
          "null"
        ]
      },
      "uid": {
        "description": "UID of the resource. (when there is a single resource which can be described). More info: http://kubernetes.io/docs/user-guide/identifiers#uids",
        "type": [
          "string",
          "null"
        ]
      }
    },
    "type": "object"
  },
  "tests": [
    {
      "description": "llama 70b generated valid",
      "valid": true,
      "data": {
        "causes": [
          {
            "field": "name",
            "message": "The name is already in use",
            "reason": "AlreadyExists"
          },
          {
            "field": "items[0].name",
            "message": "The name is too long",
            "reason": "Invalid"
          }
        ],
        "group": "apps",
        "kind": "Deployment",
        "name": "my-deployment",
        "retryAfterSeconds": 30,
        "uid": "12345678-1234-1234-1234-123456789012"
      }
    },
    {
      "description": "llama-70b generated negative; focus on type keyword with an array of types",
      "valid": false,
      "rust_error": "\"not an array\" is not of types \"array\", \"null\"",
      "python_error": "'not an array' is not of type 'array', 'null'\n\nFailed validating 'type' in schema['properties']['causes']:\n    {'description': 'The Causes array includes more details associated '\n                    'with the StatusReason failure. Not all StatusReasons '\n                    'may provide detailed causes.',\n     'items': {'$ref': '#/definitions/https:__kubernetesjsonschema_dev_master__defs_json_defs_io_k8s_apimachinery_pkg_apis_meta_v1_StatusCause'},\n     'type': ['array', 'null']}\n\nOn instance['causes']:\n    'not an array'",
      "data": {
        "causes": "not an array",
        "group": "apps",
        "kind": "Deployment",
        "name": "my-deployment",
        "retryAfterSeconds": 30,
        "uid": "12345678-1234-1234-1234-123456789012"
      }
    },
    {
      "description": "llama-70b generated negative",
      "valid": false,
      "rust_error": "123 is not of type \"string\"",
      "python_error": "123 is not of type 'string'\n\nFailed validating 'type' in schema['properties']['causes']['items']['properties']['reason']:\n    {'description': 'A machine-readable description of the cause of the '\n                    'error. If this value is empty there is no information '\n                    'available.',\n     'type': 'string'}\n\nOn instance['causes'][1]['reason']:\n    123",
      "data": {
        "causes": [
          {
            "field": "name",
            "message": "The name is already in use",
            "reason": "AlreadyExists"
          },
          {
            "field": "items[0].name",
            "message": "The name is too long",
            "reason": 123
          }
        ],
        "group": "apps",
        "kind": "Deployment",
        "name": "my-deployment",
        "retryAfterSeconds": 30,
        "uid": "12345678-1234-1234-1234-123456789012"
      }
    },
    {
      "description": "llama-70b generated negative; focus on type keyword with an array of types",
      "valid": false,
      "rust_error": "true is not of types \"null\", \"string\"",
      "python_error": "True is not of type 'string', 'null'\n\nFailed validating 'type' in schema['properties']['group']:\n    {'description': 'The group attribute of the resource associated with '\n                    'the status StatusReason.',\n     'type': ['string', 'null']}\n\nOn instance['group']:\n    True",
      "data": {
        "causes": [
          {
            "field": "name",
            "message": "The name is already in use",
            "reason": "AlreadyExists"
          },
          {
            "field": "items[0].name",
            "message": "The name is too long",
            "reason": "Invalid"
          }
        ],
        "group": true,
        "kind": "Deployment",
        "name": "my-deployment",
        "retryAfterSeconds": 30,
        "uid": "12345678-1234-1234-1234-123456789012"
      }
    },
    {
      "description": "llama-70b generated negative",
      "valid": false,
      "rust_error": "123 is not of type \"string\"",
      "python_error": "123 is not of type 'string'\n\nFailed validating 'type' in schema['properties']['causes']['items']['properties']['reason']:\n    {'description': 'A machine-readable description of the cause of the '\n                    'error. If this value is empty there is no information '\n                    'available.',\n     'type': 'string'}\n\nOn instance['causes'][0]['reason']:\n    123",
      "data": {
        "causes": [
          {
            "field": "name",
            "message": "The name is already in use",
            "reason": 123
          },
          {
            "field": "items[0].name",
            "message": "The name is too long",
            "reason": "Invalid"
          }
        ],
        "group": "apps",
        "kind": "Deployment",
        "name": "my-deployment",
        "retryAfterSeconds": 30,
        "uid": "12345678-1234-1234-1234-123456789012"
      }
    },
    {
      "description": "llama 70b generated positive",
      "valid": true,
      "data": {
        "causes": [
          {
            "field": "name",
            "message": "Invalid name",
            "reason": "Invalid"
          },
          {
            "field": "items[0].name",
            "message": "Invalid item name",
            "reason": "Invalid"
          }
        ],
        "group": "apps",
        "kind": "Deployment",
        "name": "example-deployment",
        "retryAfterSeconds": 30,
        "uid": "12345678-1234-1234-1234-123456789012"
      }
    },
    {
      "description": "llama-70b generated negative; focus on type keyword with an array of types",
      "valid": false,
      "rust_error": "\"Invalid value for causes, it should be an array or null\" is not of types \"array\", \"null\"",
      "python_error": "'Invalid value for causes, it should be an array or null' is not of type 'array', 'null'\n\nFailed validating 'type' in schema['properties']['causes']:\n    {'description': 'The Causes array includes more details associated '\n                    'with the StatusReason failure. Not all StatusReasons '\n                    'may provide detailed causes.',\n     'items': {'$ref': '#/definitions/https:__kubernetesjsonschema_dev_master__defs_json_defs_io_k8s_apimachinery_pkg_apis_meta_v1_StatusCause'},\n     'type': ['array', 'null']}\n\nOn instance['causes']:\n    'Invalid value for causes, it should be an array or null'",
      "data": {
        "causes": "Invalid value for causes, it should be an array or null",
        "group": "apps",
        "kind": "Deployment",
        "name": "example-deployment",
        "retryAfterSeconds": 30,
        "uid": "12345678-1234-1234-1234-123456789012"
      }
    }
  ]
}